sensors:
- name: top_kinect
  type: kinect
  source: /amigo/top_kinect/rgbd
  frame_id: /amigo/top_kinect/openni_rgb_optical_frame
  association_modules:
  - type: PointNormal
    parameters:
        association_correspondence_distance: 0.1
        position_weight: 1.0
        curvature_weight: 2.0
        visualize: 1
  - type: PolygonHeight
    parameters:
        cell_size: 0.03
        max_range: 3.0
        tolerance: 0.05
        min_cluster_size: 20
        visualize: 1
  segmentation_modules:
  - type: EuclideanClustering
    parameters:
        tolerance: 0.05
        min_cluster_size: 20
        visualize: 1

visualize: 1

perception:
    fit_shapes: 1
    modules:
#    -  lib: libsize_matcher.so
#    -  lib: libodu_finder.so
#    -  lib: libhuman_contour_matcher.so

#plugins:
#- name: hello_world
#  lib: libed_hello_world_plugin.so
#  parameters:
#      text: "Hello world! 1"
#- name: hello_world2
#  lib: libed_hello_world_plugin.so
#  parameters:
#      text: "Hello world! 2"
#- name: hello_world3
#  lib: libed_hello_world_plugin.so
#  parameters:
#      text: "Hello world! 3"

gui:
    enabled: 1
    srv_get_measurements: /ed/gui/get_measurements
    srv_set_label: /ed/gui/set_label
    srv_raise_event: /ed/gui/raise_event
    srv_get_command: /ed/gui/get_gui_command
    map_image_topic: /ed/gui/map_image
    map_image_width: 500
    map_image_height: 500
    world_width: 11.0
    world_height: 11.0
    cam_x: 4.5
    cam_y: 4.4
    cam_z: 10.0

map_publisher:
    frequency: 10
    size_x: 20.0
    size_y: 20.8
    resolution: 0.05
    origin_x: -6.5
    origin_y: -6.5
    topic: /ed_map
    frame_id: /map
